# HG changeset patch
# User Rik <rik@octave.org>
# Date 1489036314 28800
#      Wed Mar 08 21:11:54 2017 -0800
# Node ID 99ee4dd83ceb49edcc6d6e76e3f9188613bde7a5
# Parent  aaf91b4f48e497da20917226e138c91658e7c500
dir.m: Fix regression in 529c6d0c6684.

* dir.m: Correct regular expression to separate directory from file.ext part of
name.  Change regular expression to use all file separators, as may be necessary
on Windows.

diff --git a/scripts/miscellaneous/dir.m b/scripts/miscellaneous/dir.m
--- a/scripts/miscellaneous/dir.m
+++ b/scripts/miscellaneous/dir.m
@@ -106,35 +106,35 @@ function retval = dir (directory)
       flst = readdir (flst{1});
       nf = numel (flst);
       flst = strcat ([fn filesep], flst);
     endif
   endif
 
   if (numel (flst) > 0)
 
-    fs = regexptranslate ("escape", filesep);
-    re = sprintf ('(?:^|^.+[%s])([^%s.]*.[^%s]*)?$', fs, fs, fs);
+    fs = regexptranslate ("escape", filesep ("all"));
+    re = sprintf ('(?:^.+[%s])([^%s.]*)([.][^%s]*)?$', fs, fs, fs);
     ## Collect results.
     for i = nf:-1:1
       fn = flst{i};
       [st, err, msg] = lstat (fn);
       if (err < 0)
         warning ("dir: 'lstat (%s)' failed: %s", fn, msg);
       else
         ## If we are looking at a link that points to something,
         ## return info about the target of the link, otherwise, return
         ## info about the link itself.
         if (S_ISLNK (st.mode))
           [xst, err, msg] = stat (fn);
           if (! err)
             st = xst;
           endif
         endif
-        fn = regexprep (fn, re, '$1');
+        fn = regexprep (fn, re, '$1$2');
         info(i,1).name = fn;
         lt = localtime (st.mtime);
         info(i,1).date = strftime ("%d-%b-%Y %T", lt);
         info(i,1).bytes = st.size;
         info(i,1).isdir = S_ISDIR (st.mode);
         info(i,1).datenum = [lt.year + 1900, lt.mon + 1, lt.mday, ...
                              lt.hour, lt.min, lt.sec];
         info(i,1).statinfo = st;
